picky/src/x509/pkcs7/ctl.rs

```diff
@@ -0,0 +1,99 @@
+use picky_asn1_x509::content_info::ContentValue;
+use picky_asn1_x509::pkcs7::ctl::CTLEntry;
+use std::io::{self, Cursor, Read};
+
+use crate::x509::pkcs7::{Pkcs7, Pkcs7Error};
+use thiserror::Error;
+
+pub use picky_asn1_x509::pkcs7::ctl::CTLEntryAttributeValues;
+
+#[derive(Debug, Error)]
+pub enum CtlError {
+    #[error("Failed to download CTL:  {description}")]
+    DownloadError { description: String },
+    #[error("{description}")]
+    ExtractingError { description: String },
+    #[error("Failed to parse CertificateTrustList: {0}")]
+    FailedToParseCtl(Pkcs7Error),
+    #[error(transparent)]
+    IoError(#[from] io::Error),
+    #[error("For CTL we expect CertificateTrustList content in EncapsulatedContentInfo, but something else")]
+    IncorrectContentValue,
+}
+
+pub struct CertificateTrustList {
+    pkcs7: Pkcs7,
+}
+
+impl CertificateTrustList {
+    pub fn new() -> Result<Self, CtlError> {
```

Comment:
Rename to make it a bit more clear the operation is costly
```suggestion
    pub fn fetch() -> Result<Self, CtlError> {
```